llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -7978,6 +7978,12 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
     return;
   }
   case Intrinsic::experimental_vector_partial_reduce_add: {
+
+    if(!TLI.shouldExpandPartialReductionIntrinsic(&I))) {
+      visitTargetIntrinsic(I, Intrinsic);
```

Comment:
Needs codegen test?

Reply:
I see, but does this change need to be part of the LV changes? And would it be possible test this change separately?

Reply:
I think it should be possible to remove this change from this PR so I'll give that a go.

Reply:
On second thought, removing this change from this PR would mean adding it to the AArch64 lowering PR which is not the cleanest, or creating a trivially small patch that adds the target hook, which I don't think is worth it, so I think this needs to stay in here.

Reply:
Not sure if I've misunderstood but the TargetLowering changes only relate to code generation whereas the meat of this patch relates to a LoopVectorize transformation.

To me moving the TargetLowering change into the PR that exercise the change (currently this test is dead code with `visitTargetIntrinsic` being unreachable) into the PR that implements the `visitTargetIntrinsic` side of the branch is the way to go.

Reply:
That makes sense to me, I've removed it now and will fold it into a separate PR.